### ISSUE TITLE
Remove rex_pcre dependency from core

### ISSUE THREAD

**Issue by thefosk** [Mashape/kong#167]
Use Lua patterns instead.

**Comment by thibaultCha** [comment]
:+1:

**Comment by thibaultCha** [comment]
I am actually not sure if this is going to be possible. Places where we need a regex, checked if a test case with a working pattern is provided:

- [ ] Validate a `public_dns`
- [x] Validate a uuid for entities in the Cassandra DAO
- [ ] (needed soon) catching invalid `target_url` #149 
- [ ] Any future other plugin that will have to rely on regexes
- [ ] Some places in the specs

uuid pattern test case (I do remember using a regex instead, so we'll need to investigate if it covers 100%):

```lua
local x = "%x"
local pattern = { x:rep(8), x:rep(4), x:rep(4), x:rep(4), x:rep(12) }
pattern = table.concat(pattern, '%-')

local guid = "3F2504E0-4F89-41D3-9A0C-0305E82C3301"
assert(guid:match(pattern))

local guid = "fba88588-ffd0-4769-c782-7c537c92078d"
assert(guid:match(pattern))

local guid = "3F2504E0-4F89-41D3-9A0C0305E82C3301"
assert(not guid:match(pattern))

local guid = "304E0-4F89-41D3-9A0C0305E8^3301"
assert(not guid:match(pattern))

local guid = "3F2504E0-4F89-41D3-9A0§-03(5E82C3301"
assert(not guid:match(pattern))
```

**Comment by thefosk** [comment]
If if it's not feasible to remove it from everywhere, we shouldn't force it. Just reduce regex usage to the places where it's absolutely necessary would be okay.

**Comment by thibaultCha** [comment]
It might be possible, we might also use a slightly more enhanced way of using the patterns (like a library: http://www.inf.puc-rio.br/~roberto/lpeg/lpeg.html)

**Comment by thibaultCha** [comment]
Just updated the issue with a working version of a `public_dns` check. Please do test it with other values you could think of.

**Comment by thibaultCha** [comment]
Updated for the specs part too

**Comment by thibaultCha** [comment]
Merged #167 which does some improvements and use patterns in most places instead of regexes. PCRE is not removed and the `regex` field is still available in schema definitions.
